src/doom-shell.html

```diff
@@ -329,9 +332,47 @@
             // For all other keys, use the original function
             return originalLookupKeyCodeForEvent(event);
           };
+
+          // In the preRun stage, mount IDBFS so that it’s available to DOOM right away.
+          // This enables syncing browser storage to doom_saves folder
+          if (!FS.analyzePath("/doom_saves").exists) {
+            // Ensure the directory exists BEFORE mounting IDBFS
+            FS.mkdir("/doom_saves");
+          }
+
+          FS.mount(IDBFS, {}, "/doom_saves");
+
+          console.log("Mounted IDBFS, syncing from IndexedDB...");
+
+          FS.syncfs(true, function (err) {
+            if (err) {
+              console.error("Failed to sync from IndexedDB:", err);
+            } else {
+              console.log("Successfully loaded saved data from IndexedDB.");
+              console.log("Files after sync:", FS.readdir("/doom_saves"));
+            }
+          });
         },
       ],
       postRun: [],
+      onSaveGame: function () {
```

Comment:
see onSaveGame hook on siis see mille ma ise juurde lõin. See sünkroniseerib mängu storagei ja browseri andmebaasi (indexedDB), et uuendada andmebaasis olevaid salvestatud mänge. Seda hooki kutsun iga kord välja kui keegi mängu sees seivib geimi, siis on kõige suurem tõenäosus, et kõik on ap-tu-déit ja mõni seiv geim ei lähe kaduma